src/boot/efi/boot.c

```diff
@@ -28,9 +28,27 @@ enum loader_type {
         LOADER_UNDEFINED,
         LOADER_EFI,
         LOADER_LINUX,
+        LOADER_SIGNATURE_KEYS,
 };
 
 typedef struct {
+        const CHAR16 *name;
+        const CHAR16 *filename;
+        const EFI_GUID vendor;
+        CHAR8 *buffer;
+        UINTN size;
+} SecureBootVariable;
```

Comment:
so you have a single struct and use different parts of it in different contexts. That's strange... should be two structs then. (and quite frankly i wouldn't bother with the buffer/size part here, and load them right before we need them, i.e. keep the vars for it on the stackframe